File: src/app/auth-components/shared/sign-up-dto.ts
export interface SignupData {
  // Step 1
  email: string;
  name: string;
  password: string;

  // Step 2
  country: string;
  state: string;
  avatar: string;

  // Step 3
  skills: Array<{
    name: string;
    level: string;
  }>;

  // Step 4
  languages: string[];
  bio: string;
}

File: src/main/java/com/selflearning/dtos/SignupDto.java
package com.selflearning.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.util.List;

@Data
public class SignupDto {
    private String name;
    private String email;
    private String password;
    @JsonProperty("photoUrlSeed")
    private String photoUrl;
    private String designation;
    private String location;
    private List<Skill> skills;
    private String bio;
}

